libs/remote_storage/tests/common/tests.rs

```diff
@@ -265,6 +397,80 @@ async fn delete_objects_works(ctx: &mut MaybeEnabledStorage) -> anyhow::Result<(
     Ok(())
 }
 
+/// Tests that delete_prefix() will delete all objects matching a prefix, including
+/// partial prefixes (i.e. "/foo" matches "/foobar").
+#[test_context(MaybeEnabledStorageWithSimpleTestBlobs)]
```

Comment:
This only tests with the azure client, right? Would be nice to test with s3 as well, but we do that in integration tests, so there's some coverage at least.